students/allen_maxwell/Lesson_05/database.py

```diff
@@ -0,0 +1,104 @@
+'''Imports csv file to MongoDb'''
+
+import csv
+import os
+import logging
+from pymongo import MongoClient
+from pymongo import errors as pyerror
+
+logging.basicConfig(filename="db.log", filemode="w", level=logging.DEBUG)
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+class MongoDBConnection():
+    '''MongoDB Connection'''
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        '''Initiates the connection protocol'''
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        '''Connects to the database client'''
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        '''Exits the connection'''
+        self.connection.close()
+
+def import_data(directory_name, product_file, customer_file, rentals_file):
+    '''
+    This function takes a directory name three csv files as input (product data,
+    customer data, and rentals data). It returns 2 tuples with a record count of the
+    number of data and errors for each file (product, customer, rentals).
+    '''
+    LOGGER.debug('Importing csv file')
+    mongo = MongoDBConnection()
+    with mongo:
+        database = mongo.connection.hp_norton
+
+         # Create collections
+        products = database[product_file]
+        customers = database[customer_file]
+        rentals = database[rentals_file]
+
+        products.drop
+        customers.drop
+        rentals.drop
+
+        prod_count, prod_errors = import_csv(directory_name, product_file, products)
+        cust_count, cust_errors = import_csv(directory_name, customer_file, customers)
+        rent_count, rent_errors = import_csv(directory_name, rentals_file, rentals)
+    return((prod_count, cust_count, rent_count), (prod_errors, cust_errors, rent_errors))
+
+def import_csv(directory_name, file_name, database):
+    '''
+    Creates and populates a new MongoDB database with data,
+    returns the number of files and errors
+    '''
+    LOGGER.debug('Importing %s', file_name)
+    errors = 0
+    count = 0
+    try:
+        file_csv = f'{file_name}.csv'
+        with open(os.path.join(directory_name, file_csv)) as file:
+            count = len(database.insert_many(csv.DictReader(file)).inserted_ids)
+    except (pyerror.InvalidOperation, FileNotFoundError) as error:
+        LOGGER.warning('Oops! something went wrong while reading %s: %s', file_csv, error)
+        errors += 1
+    return count, errors
+
+def show_available_products():
+    '''Returns a Python dictionary of products listed as available'''
+    results = {}
+    mongo = MongoDBConnection()
+    with mongo:
+        database = mongo.connection.hp_norton
+        LOGGER.info('Searching for available products')
+        for product in database.products.find({'quantity_available': {'$gt': '0'}}):

```

Comment:
This line still uses "products" as the name of the collection, which will cause _test_show_available_products()_ to fail, because it's populating "test_products". Note that this line could also be:

`for product in database['products'].find({'quantity_available': {'$gt': '0'}}):`

Which would allow you to pass the name of the collection as a parameter (replacing 'products'). This is probably passing on your side because your _products_ collection already has the same entries that your tests are looking for.